### PR TITLE
Continuation tokens for searches

### PR DESCRIPTION
**Pull request by olaughter**
Adds the ability to get the next set of results. Either at the family level, i.e. 'get more families from the search I just ran', or at the passage level, i.e. 'get more passages for a family from the search I just ran'. It works using [continuation tokens](CCLW.family.4934.0) to get more of grouped results from the initial query. And enables the ability to paginate to the end of the results set, for either families or passages.

Vespas underlying implementation of continuation tokens uses the bucket storage as markers. This means that the query is fully reproducible as long as the data does not change. I've been able to confirm this is the case in practice by rerunning a continuation after a day and getting the exact same results.

There was a partial interface previously that accepted a single token, this pr builds on that to also allow passage level continuations, as well as returning the tokens to the user. Note that this breaks the continuation token interface previously in place as the parameters object no longer accepts a string.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,19 +101,58 @@ request = SearchParameters(
...
+
+For the next selection of families:
+
+```python
+response = adaptor.search(SearchParameters(query_string="forest fires"))
+
+follow_up_request = SearchParameters(
+    query_string="forest fires"
+    continuation_tokens=[response.continuation_token],
+
```
Comment (THOR300) [inline README.md:131]: Empty line here and on line 150

File: tests/test_search_adaptors.py [hunk 2]
```diff
@@ -123,3 +125,162 @@ def test_vespa_search_adaptor__sensitive(fake_vespa_credentials):
...
+
+    this_continuation = initial_response.this_continuation_token
+    passage_continuation = initial_response.families[0].continuation_token
+
+    # Passage Increment
+    request = SearchParameters(
+        query_string=query_string,
+        limit=limit,
+        max_hits_per_family=max_hits_per_family,
+        continuation_tokens=[this_continuation, passage_continuation],
```
Comment (THOR300) [inline tests/test_search_adaptors.py:219]: Out of interest if these are just a list of strings how does vespa know which one is in reference to the families or passages? Is it just a clever format that vespa can understand? 

Just thinking if the order of the tokens changed would this still work?

**Review by THOR300 — APPROVED** [review]
LGTM !

File: README.md (hunk 1 above)
Comment (olaughter) [reply README.md:131]: Fixed

File: tests/test_search_adaptors.py (hunk 2 above)
Comment (olaughter) [reply tests/test_search_adaptors.py:219]: Example token at the family level: `BGAAABGBCBC`
And another at the passage level: `BKAAAAABKCBEBC`

So a family and a passage can go together in any order, (maybe they use the length of the string to determine it — I don't actually know).

As a side note, the ordering does become important if both tokens are from the same level of grouping. The docs say its only the [last one thats used](https://docs.vespa.ai/en/reference/grouping-syntax.html#:~:text=This%20is%20because%20continuations%20are%20processed%20in%20the%20order%20given%2C%20and%20they%20replace%20whatever%20continuations%20they%20collide%20with.)

**Review by katybaulch — APPROVED** [review]
Really neat, great stuff! 🦖

**Pull request merged**